### PR TITLE
Feature lwdev 7729 update to net core 2.1

### PR DESCRIPTION
**Pull request by njannink**
Jira task: https://lykkex.atlassian.net/browse/LWDEV-7729

Performed tasks:
- Update to .net core 2.1
- Updated internal and external nuget packages
- Fixed build and resharper warnings
- Fixed mismatch between HFT and (new) MatchingEngine response and status codes
- Regenerated the C# autorest client

### CODE REVIEW

**Comment by polonzer** [comment]
Please divide the refactoring and updating libraries into separate PRs.

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -89,14 +88,7 @@ public class OrdersController : Controller
...
                 case OrderStatus.Rejected:
-                    orders = orders.Where(order =>
-                        order.Status == Core.Domain.OrderStatus.NotEnoughFunds
-                        || order.Status == Core.Domain.OrderStatus.NoLiquidity
-                        || order.Status == Core.Domain.OrderStatus.UnknownAsset
-                        || order.Status == Core.Domain.OrderStatus.LeadToNegativeSpread
-                        || order.Status == Core.Domain.OrderStatus.ReservedVolumeGreaterThanBalance
-                        || order.Status == Core.Domain.OrderStatus.TooSmallVolume
-                        || order.Status == Core.Domain.OrderStatus.Runtime);
+                    orders = orders.Where(order => order.Status.IsRejected());
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Controllers/OrdersController.cs]: This code will not work. This method cannot be parsed by MongoDB driver.

File: src/Lykke.Service.HFT.Wamp/Consumers/Messages/LimitOrderMessage.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class Order
             public string AssetPairId { get; set; }
             public string ClientId { get; set; }
             public double Volume { get; set; }
-            public OrderStatus Status { get; set; }
+            public string Status { get; set; }
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Wamp/Consumers/Messages/LimitOrderMessage.cs:24]: I suppose the related code should be changed too.

File: src/Lykke.Service.HFT.Core/Guard.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        ///     Throws <see cref="ArgumentNullException" /> when a <paramref name="param" /> is null.
+        /// </exception>
+        /// <param name="param">
+        ///     The object to validate.
+        /// </param>
+        /// <param name="paramName">
+        ///     The name of the validating parameter.
+        /// </param>
+        [ContractAnnotation(@"param:null => halt")]
+        public static void AgainstNull(object param, [Localizable(false)] string paramName)
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Core/Guard.cs]: Please don't use guard classes, at least not introduse handwritten ones.
http://www.paraesthesia.com/archive/2011/09/30/six-reasons-not-to-use-guard-classes.aspx/

File: src/Lykke.Service.HFT.Services/Consumers/Messages/LimitOrderMessage.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ public class Order
             public string AssetPairId { get; set; }
             public string ClientId { get; set; }
             public double Volume { get; set; }
-            public OrderStatus Status { get; set; }
+            public string Status { get; set; }
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/Consumers/Messages/LimitOrderMessage.cs:24]: I suppose the related code should be changed too.

**Comment by njannink** [comment]
Couldn't split refactorying and updating to .netcore 2.1 into seperate PR since after updating framework and nuget packages the HFT service was starting to give exceptions.

File: src/Lykke.Service.HFT.Services/Consumers/Messages/LimitOrderMessage.cs (hunk 4 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Services/Consumers/Messages/LimitOrderMessage.cs:24]: This OrderStatus was only used as DTO for deserializing. So a string is sufficient and then the additional enum doesn't need to be maintained.

File: src/Lykke.Service.HFT.Wamp/Consumers/Messages/LimitOrderMessage.cs (hunk 2 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Wamp/Consumers/Messages/LimitOrderMessage.cs:24]: This OrderStatus was only used as DTO for deserializing. So a string is sufficient and then the additional enum doesn't need to be maintained.

File: src/Lykke.Service.HFT.Core/Guard.cs (hunk 3 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Core/Guard.cs]: ok removed again. I know about these rules but it's more a matter of taste.

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs (hunk 1 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Controllers/OrdersController.cs]: inverted the check so newly added statussus will end up as rejected.

**Review by polonzer — APPROVED** [review]

**Comment by njannink** [comment]
Test on dev passed can be merged

**Pull request merged**